tkg/managementcomponents/helper.go

```diff
@@ -44,7 +52,7 @@ func GetTKGPackageConfigValuesFileFromUserConfig(managementPackageVersion string
 	default:
 		return "", errors.Errorf("unknown provider type %q", providerType)
 	}
-
+	addonsManagerPackageVersion := correctAddonsManagerPackageVersion(managementPackageVersion)
```

Comment:
The managementPackageVersion will be empty unless set by user. We need to get it from the BOM. I dont see it being done here or in the caller function.

Also please add an env for ADDONS_MANAGER_PACKAGE_VERSION to override the default that we got from BOM

Reply:
moved all of this logic to calling function (getTKGPackageConfigValuesFile) because we'll need the tkgclient to get the version from the bom file